ui/app/routes/vault/cluster/clients/index.js

```diff
@@ -1,85 +1,40 @@
 import Route from '@ember/routing/route';
-import ClusterRoute from 'vault/mixins/cluster-route';
-import { hash } from 'rsvp';
-
-export default Route.extend(ClusterRoute, {
-  queryParams: {
-    tab: {
-      refreshModel: true,
-    },
-    start: {
-      refreshModel: true,
-    },
-    end: {
-      refreshModel: true,
-    },
-  },
-
-  async getLicense() {
-    try {
-      return await this.store.queryRecord('license', {});
-    } catch (e) {
-      // ARG TODO handle
-      return e;
-    }
-  },
-
-  async getActivity(start_time) {
-    try {
-      return await this.store.queryRecord('clients/activity', { start_time });
-    } catch (e) {
-      // ARG TODO handle
-      return e;
-    }
-  },
-
-  async getMonthly() {
-    try {
-      return await this.store.queryRecord('clients/monthly', {});
-    } catch (e) {
-      // ARG TODO handle
-      return e;
-    }
-  },
-
-  rfc33395ToMonthYear(timestamp) {
-    // return ['2021', 2] (e.g. 2021 March, make 0-indexed)
-    return timestamp
-      ? [timestamp.split('-')[0], Number(timestamp.split('-')[1].replace(/^0+/, '')) - 1]
-      : null;
-  },
-
-  async model() {
-    let config = this.store.queryRecord('clients/config', {}).catch((e) => {
-      console.debug(e);
-      // swallowing error so activity can show if no config permissions
-      return {};
+import RSVP from 'rsvp';
+import { action } from '@ember/object';
+
+export default class ClientsRoute extends Route {
+  // async getMonthly() {
```

Comment:
[RSVP.hash](https://guides.emberjs.com/release/routing/specifying-a-routes-model/#toc_multiple-models) handles multiple API requests - so I don't see the need to separate these functions into separate `async` calls, but perhaps I'm missing something?

Reply:
Yeah `config` is required to check if tracking is enabled, and without `monthly` there is no data to display. So if I understand correctly. We would want to trigger a more general error state of neither of those promises were fulfilled.